docs/msi_v2/guidance_for_sdks_consuming_msal.md

```diff
@@ -0,0 +1,62 @@
+# Guidance for SDKs Consuming MSAL
+
+## Overview
+
+To support MSI V2 authentication with the `/credential` endpoint, the **Azure SDK** leverages the `IMsalMtlsHttpClientFactory` interface and **certificate management APIs** for secure communication with Azure AD using **mutual TLS (mTLS)**.
+
+This section covers:
+- How Azure SDK uses **`IMsalMtlsHttpClientFactory`** for MTLS authentication.
+- How SDKs interact with the **certificate APIs** to obtain the binding certificate certificates.
+- The **new `CertificateRefreshed` event**, which notifies when a binding certificate is updated.
+
+---
+
+## **Handling Certificate Rotation for Long-Lived Clients**
+
+The Problem: Certificate Expiry and Rotation
+
+- mTLS Proof of Possession (PoP) tokens are signed by a binding certificate.
+- The binding certificate is valid for 90 days.
+- A new certificate is made available 5 days before expiration.
+- The SDKs consuming MSAL (customizing httpclient) must ensure that its HttpClient uses the latest certificate.
+
+Proposed Solution
+- SDK clients must monitor certificate updates and refresh their HttpClient dynamically.
+- MSAL exposes an event-driven model to notify when the binding certificate is refreshed.
+
+## **New Interface: `IMsalMtlsHttpClientFactory`**
+
+MSAL introduces the `IMsalMtlsHttpClientFactory` interface to facilitate **mTLS-based authentication** in Azure SDKs. This ensures:
+- Secure token acquisition by enabling **mTLS authentication**.
+- **Reusable `HttpClient` instances** to prevent socket exhaustion.
+- Optimized **Azure SDK integration** for managing identities securely.
+
+### **Interface Definition**
+
+```csharp
+public interface IMsalMtlsHttpClientFactory : IMsalHttpClientFactory
```

Comment:
I'm thinking that it may be useful to have this as an overload to `WithHttpClientFactory` and that either or both can be called on an application builder. This would mean that it's possible for callers to get called on the appropriate callback depending on if the auth flow is mtls or not without having to initialize 2 differnt application builders (one for bearer and one for mtls flows)